Source/WebCore/rendering/RenderGrid.h

```diff
@@ -129,6 +129,9 @@ class RenderGrid final : public RenderBlock {
 
     LayoutUnit masonryContentSize() const;
     Vector<LayoutRect> gridItemsLayoutRects();
+
+    bool shouldCheckExplicitIntrinsicInnerLogicalSize(GridTrackSizingDirection) const;
```

Comment:
shouldn't  `friend class GridTrackSizingAlgorithm;` make it not required?

Reply:
IndefiniteSizeStrategy calls this method, which is in GridTrackSizingAlgorithm file, but other than that is unrelated to the friend class reference.